### PR TITLE
Add directions/commands for how to run each WordCount example with each runner.

### PR DESCRIPTION
**Pull request by hadarhg**
**Minimal WC**
Need to check 'How to run' directions for:
- [x] Java: Direct
- [ ] Java: Apex
- [x] Java: Flink-local
- [x] Java: Flink-cluster
- [ ] Java: Spark
- [ ] Java: Dataflow
- [x] Python: Direct
- [x] Python: Dataflow

**WC**
Need to check 'How to run' directions for:
- [x] Java: Direct
- [x] Java: Apex
- [x] Java: Flink-local
- [x] Java: Flink-cluster
- [x] Java: Spark
- [x] Java: Dataflow
- [x] Python: Direct
- [x] Python: Dataflow
*Should all be correct as these were copied from the quickstarts

**Debugging WC**
Need to check 'How to run' directions for:
- [ ] Java: Direct
- [ ] Java: Apex
- [x] Java: Flink-local
- [x] Java: Flink-cluster
- [ ] Java: Spark
- [ ] Java: Dataflow
- [x] Python: Direct
- [x] Python: Dataflow

**Windowed WC**
Need to check 'How to run' directions for:
- [ ] Java: Direct
- [ ] Java: Apex
- [x] Java: Flink-local
- [x] Java: Flink-cluster
- [ ] Java: Spark
- [ ] Java: Dataflow

### CODE REVIEW

File: src/get-started/wordcount-example.md [hunk 1]
```diff
@@ -396,7 +692,7 @@ public static void main(String[] args) throws IOException {
 ```
 
 ```py
-This feature is not yet available in the Beam SDK for Python.
+This feature is not yet available ifor the Python SDK.
```
Comment (aaltay) [inline src/get-started/wordcount-example.md]: `ifor` -> `for`

File: src/get-started/wordcount-example.md [hunk 2]
```diff
@@ -31,7 +31,90 @@ Each WordCount example introduces different concepts in the Beam programming mod
 
 Minimal WordCount demonstrates a simple pipeline that can read from a text file, apply transforms to tokenize and count the words, and write the data to an output text file. This example hard-codes the locations for its input and output files and doesn't perform any error checking; it is intended to only show you the "bare bones" of creating a Beam pipeline. This lack of parameterization makes this particular pipeline less portable across different runners than standard Beam pipelines. In later examples, we will parameterize the pipeline's input and output sources and show other best practices.
 
-To run this example, follow the instructions in the [Beam Examples README](https://github.com/apache/beam/blob/master/examples/java/README.md#building-and-running). To view the full code, see **[MinimalWordCount](https://github.com/apache/beam/blob/master/examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java).**
+**To run this example in Java:**
+
+{:.runner-direct}
+```
+$ mvn compile exec:java -Dexec.mainClass=org.apache.beam.examples.MinimalWordCount \
+     -Dexec.args="--inputFile=pom.xml --output=counts" -Pdirect-runner
```
Comment (aaltay) [inline src/get-started/wordcount-example.md]: Are you sure that MinimalWordCount supports input, output or non local runners: (https://github.com/apache/beam/blob/master/examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java ?)

File: src/get-started/wordcount-example.md [hunk 3]
```diff
@@ -31,7 +31,90 @@ Each WordCount example introduces different concepts in the Beam programming mod
...
+     -Pdataflow-runner
+```
+
+To view the full code in Java, see **[MinimalWordCount](https://github.com/apache/beam/blob/master/examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java).**
+
+**To run this example in Python:**
+
+{:.runner-direct}
+```
+python -m apache_beam.examples.wordcount --input README.md --output counts
```
Comment (aaltay) [inline src/get-started/wordcount-example.md]: `wordcount` -> `wordcount_minimal`

File: src/get-started/wordcount-example.md [hunk 4]
```diff
@@ -31,7 +31,90 @@ Each WordCount example introduces different concepts in the Beam programming mod
...
+{:.runner-spark}
+```
+This runner is not yet available for the Python SDK.
+```
+
+{:.runner-dataflow}
+```
+# As part of the initial setup, install Google Cloud Platform specific extra components.
+pip install apache-beam[gcp]
+python -m apache_beam.examples.wordcount --input gs://dataflow-samples/shakespeare/kinglear.txt \
```
Comment (aaltay) [inline src/get-started/wordcount-example.md]: This is a `wordcount` example. Doing this with `wordcount_minimal` requires code changes. (See CHANGE comments in https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/wordcount_minimal.py)

File: src/get-started/wordcount-example.md [hunk 5]
```diff
@@ -31,7 +31,87 @@ Each WordCount example introduces different concepts in the Beam programming mod
...
+{:.runner-spark}
+```
+This runner is not yet available for the Python SDK.
+```
+
+{:.runner-dataflow}
+```
+# As part of the initial setup, install Google Cloud Platform specific extra components.
+pip install apache-beam[gcp]
+python -m apache_beam.examples.wordcount_minimal
```
Comment (aaltay) [inline src/get-started/wordcount-example.md]: Add a note that, the code needs to be change to run with DataflowRunner. (Inside the code there are comments such as "CHANGE 1/5" ...)

**Comment by aljoscha** [comment]
@hadarhg The snippets for running on the Flink runner look good. 👍

File: src/get-started/wordcount-example.md [hunk 6]
```diff
@@ -27,11 +27,93 @@ Each WordCount example introduces different concepts in the Beam programming mod
 * **Debugging WordCount** introduces logging and debugging practices.
 * **Windowed WordCount** demonstrates how you can use Beam's programming model to handle both bounded and unbounded datasets.
 
+> Note: The instructions on this page, for how to run the WordCount examples, have not yet been verified for all runners. (See the Jira issues for the [direct](https://issues.apache.org/jira/browse/BEAM-2348), [Apex](https://issues.apache.org/jira/browse/BEAM-2349), [Spark](https://issues.apache.org/jira/browse/BEAM-2350), and [Dataflow](https://issues.apache.org/jira/browse/BEAM-2351) runners).
```
Comment (dhalperi) [inline src/get-started/wordcount-example.md]: Remove this paragraph.

File: src/get-started/wordcount-example.md [hunk 7]
```diff
@@ -27,11 +27,93 @@ Each WordCount example introduces different concepts in the Beam programming mod
...
 
 Minimal WordCount demonstrates a simple pipeline that can read from a text file, apply transforms to tokenize and count the words, and write the data to an output text file. This example hard-codes the locations for its input and output files and doesn't perform any error checking; it is intended to only show you the "bare bones" of creating a Beam pipeline. This lack of parameterization makes this particular pipeline less portable across different runners than standard Beam pipelines. In later examples, we will parameterize the pipeline's input and output sources and show other best practices.
 
-To run this example, follow the instructions in the Quickstart for [Java]({{ site.baseurl }}/get-started/quickstart-java) or [Python]({{ site.baseurl }}/get-started/quickstart-py). To view the full code, see **[MinimalWordCount](https://github.com/apache/beam/blob/master/examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java).**
+**To run this example in Java:**
+
+{:.runner-direct}
+```
+To run this pipeline, first edit all places hard-coded in the example code that are marked with a CHANGE comment. Then, you can run the pipeline by using the following command:
```
Comment (dhalperi) [inline src/get-started/wordcount-example.md]: Shouldn't this text be above, where it is not runner-specific?

Comment (dhalperi) [inline src/get-started/wordcount-example.md]: And not inside the ``` so that it does not appear in code font.

File: src/get-started/wordcount-example.md [hunk 8]
```diff
@@ -139,7 +219,7 @@ The Minimal WordCount pipeline contains five transforms:
 
 4.  The next transform formats each of the key/value pairs of unique words and occurrence counts into a printable string suitable for writing to an output file.
 
-	The map transform is a higher-level composite transform that encapsulates a simple `ParDo`. For each element in the input `PCollection`, the map transform applies a function that produces exactly one output element.
+    The map transform is a higher-level composite transform that encapsulates a simple `ParDo`; for each element in the input `PCollection`, the map transform applies a function that produces exactly one output element.
```
Comment (dhalperi) [inline src/get-started/wordcount-example.md]: why this change?

File: src/get-started/wordcount-example.md (hunk 3 above)
Comment (dhalperi) [inline src/get-started/wordcount-example.md]: @aaltay why? wordcount minimal is above.

File: src/get-started/wordcount-example.md [hunk 9]
```diff
@@ -300,7 +546,15 @@ The following sections explain these key concepts in detail, and break down the
...
-Each runner may choose to handle logs in its own way.
+Each runner may choose to handle logs in its own way. 
+
+#### Direct Runner
+
+If you execute your pipeline using `DirectRunner`, it will print the log messages directly to your local console.
+
+#### Dataflow Runner
+
+If you execute your pipeline using `DataflowRunner`, you can use Google Cloud Logging. Google Cloud Logging (currently in beta) aggregates the logs from all of your Dataflow job's workers to a single location in the Google Cloud Platform Console. You can use Cloud Logging to search and access the logs from all of the Compute Engine instances that Dataflow has spun up to complete your Dataflow job. You can add logging statements into your pipeline's `DoFn` instances that will appear in Cloud Logging as your pipeline runs.
```
Comment (dhalperi) [inline src/get-started/wordcount-example.md]: is all this text accurate, @swegner or @bjchambers? I thought it was renamed from Cloud Logging and perhaps is not in beta.

Comment (dhalperi) [inline src/get-started/wordcount-example.md]: Indeed, this is a conscious shift back to Cloud Logging from Stackdriver Logging. is that intentional, @hadarhg ?

File: src/get-started/wordcount-example.md [hunk 10]
```diff
@@ -345,18 +590,14 @@ The default Dataflow worker logging configuration can be overridden by specifyin
...
 > **Note:** This section is yet to be added. There is an open issue for this ([BEAM-791](https://issues.apache.org/jira/browse/BEAM-791)).
 
-#### Apache Apex Runner
-
-> **Note:** This section is yet to be added. There is an open issue for this ([BEAM-2285](https://issues.apache.org/jira/browse/BEAM-2285)).
-
 ### Testing your Pipeline via PAssert
 
-`PAssert` is a set of convenient PTransforms in the style of Hamcrest's collection matchers that can be used when writing Pipeline level tests to validate the contents of PCollections. `PAssert` is best used in unit tests with small data sets, but is demonstrated here as a teaching tool.
+`PAssert` is a set of convenient `PTransform`s in the style of Hamcrest's collection matchers that can be used when writing Pipeline level tests to validate the contents of PCollections. `PAssert` is best used in unit tests with small data sets, but is demonstrated here as a teaching tool.
```
Comment (dhalperi) [inline src/get-started/wordcount-example.md]: I believe removing the code font here was a conscious change in the 2.0.0 release rush. Why put it back?

File: src/get-started/wordcount-example.md [hunk 11]
```diff
@@ -345,18 +590,14 @@ The default Dataflow worker logging configuration can be overridden by specifyin
...
 
-Below, we verify that the set of filtered words matches our expected counts. Note that `PAssert` does not produce any output, and pipeline will only succeed if all of the expectations are met. See [DebuggingWordCountTest](https://github.com/apache/beam/blob/master/examples/java/src/test/java/org/apache/beam/examples/DebuggingWordCountTest.java) for an example unit test.
+Below, we verify that the set of filtered words matches our expected counts. Note that `PAssert` does not provide any output, and that successful completion of the pipeline implies that the expectations were met. See [DebuggingWordCountTest](https://github.com/apache/beam/blob/master/examples/java/src/test/java/org/apache/beam/examples/DebuggingWordCountTest.java) for an example unit test.
```
Comment (dhalperi) [inline src/get-started/wordcount-example.md]: previous text seems better; can you say more about this change?

File: src/get-started/wordcount-example.md [hunk 12]
```diff
@@ -386,24 +627,71 @@ This example, `WindowedWordCount`, counts words in text just as the previous exa
...
+Recall that the input for this example is a set of Shakespeare's texts, finite data. Therefore, this example reads bounded data from a text file:
 
 ```java
 public static void main(String[] args) throws IOException {
     Options options = ...
     Pipeline pipeline = Pipeline.create(options);
 
     PCollection<String> input = pipeline
-      .apply(TextIO.read().from(options.getInputFile()))
+      .apply(TextIO.Read.from(options.getInputFile()))
```
Comment (dhalperi) [inline src/get-started/wordcount-example.md]: now incorrect code for TextIO.read()

File: src/get-started/wordcount-example.md [hunk 13]
```diff
@@ -451,9 +735,9 @@ This feature is not yet available in the Beam SDK for Python.
 
 ### Windowing
 
-Beam uses a concept called **Windowing** to subdivide a `PCollection` according to the timestamps of its individual elements. PTransforms that aggregate multiple elements, process each `PCollection` as a succession of multiple, finite windows, even though the entire collection itself may be of infinite size (unbounded).
+Beam uses a concept called **Windowing** to subdivide a `PCollection` according to the timestamps of its individual elements. `PTransforms` that aggregate multiple elements, process each `PCollection` as a succession of multiple, finite windows, even though the entire collection itself may be of infinite size (unbounded).
```
Comment (dhalperi) [inline src/get-started/wordcount-example.md]: drop code font?

File: src/get-started/wordcount-example.md [hunk 14]
```diff
@@ -451,9 +735,9 @@ This feature is not yet available in the Beam SDK for Python.
...
 
-The `WindowedWordCount` example applies fixed-time windowing, wherein each window represents a fixed time interval. The fixed window size for this example defaults to 1 minute (you can change this with a command-line option). 
+The `WindowingWordCount` example applies fixed-time windowing, wherein each window represents a fixed time interval. The fixed window size for this example defaults to 1 minute (you can change this with a command-line option).
```
Comment (dhalperi) [inline src/get-started/wordcount-example.md]: which is the correct name? I thought it was `WindowedWordCount`. https://github.com/apache/beam/blob/master/examples/java/src/main/java/org/apache/beam/examples/WindowedWordCount.java

File: src/get-started/wordcount-example.md [hunk 15]
```diff
@@ -467,7 +751,7 @@ This feature is not yet available in the Beam SDK for Python.
 
 ### Reusing PTransforms over windowed PCollections
 
-You can reuse existing PTransforms that were created for manipulating simple PCollections over windowed PCollections as well.
+You can reuse existing `PTransform`s, that were created for manipulating simple `PCollection`s, over windowed `PCollection`s as well.
```
Comment (dhalperi) [inline src/get-started/wordcount-example.md]: same code font comments.,

File: src/get-started/wordcount-example.md [hunk 16]
```diff
@@ -479,17 +763,19 @@ This feature is not yet available in the Beam SDK for Python.
...
 ```java
-  wordCounts
-      .apply(MapElements.via(new WordCount.FormatAsTextFn()))
-      .apply(new WriteOneFilePerWindow(output, options.getNumShards()));
+wordCounts.apply(ParDo.of(new FormatAsTableRowFn()))
+    .apply(BigQueryIO.Write
+      .to(getTableReference(options))
+      .withSchema(getSchema())
+      .withCreateDisposition(BigQueryIO.Write.CreateDisposition.CREATE_IF_NEEDED)
+      .withWriteDisposition(BigQueryIO.Write.WriteDisposition.WRITE_APPEND));
```
Comment (dhalperi) [inline src/get-started/wordcount-example.md]: I think this is wrong. @kennknowles ?

File: src/get-started/wordcount-example.md (hunk 9 above)
Comment (swegner) [inline src/get-started/wordcount-example.md]: Based on [their documentation page](https://cloud.google.com/logging/docs/), it seems "Stackdriver Logging" is their preferred naming. And as @dhalperi mentions, they are out of beta.

File: src/get-started/wordcount-example.md (hunk 16 above)
Comment (kennknowles) [inline src/get-started/wordcount-example.md]: The code should remain untouched; the paragraph above it should be corrected to indicate that it writes one file per window.

**Comment by dhalperi** [comment]
Also, there's a toggle for "Adapt this page for: Java SDK | Python SDK" at the top that doesn't work.

File: src/get-started/wordcount-example.md (hunk 6 above)
Comment (hadarhg) [inline src/get-started/wordcount-example.md]: Would love to do that but unfortunately the note would then be in code font, including the link (i.e. link cannot be clicked)

**Comment by hadarhg** [comment]
@dhalperi Bad merge, sorry for the mess! Fixed it. PTAL

**Comment by dhalperi** [comment]
Thanks Hadar!
